lib/fluent/plugin/in_relp.rb

```diff
@@ -14,47 +14,78 @@ class RelpInput < Input
     desc 'The bind address to listen to.'
     config_param :bind, :string, default: '0.0.0.0'
     desc 'SSL configuration string, format certificate_path:key_path:certificate_authority_path'
-    config_param :ssl_config, :string, default: nil
+    config_param :ssl_config, :string, default: nil, deprecated: 'Use ssl_cert and ssl_ca_file instead'
+
+    config_section :ssl_cert, param_name: :ssl_certs, multi: true, required: false do
+      desc 'Path to server SSL cert'
+      config_param :cert, :string
+      desc 'Path to server SSL cert key'
+      config_param :key, :string
+      config_section :extra_cert, param_name: :extra_certs, multi: true, required: false do
+        desc 'Path to extra server SSL certs'
+        config_param :cert, :string
+      end
+    end
+    desc 'SSL ca_file for clients'
+    config_param :ssl_ca_file, :string, default: nil
+
+    @ssl_context = nil
 
     def configure(conf)
-        super
+      super
+      if !@ssl_config.nil? || !@ssl_certs.empty? || !@ssl_ca_file.nil?
+        @ssl_context = OpenSSL::SSL::SSLContext.new(:TLSv1_2)
```

Comment:
Should the tls version be hardcoded like this?  If you don't specify a tls version, will it automatically try to use the latest version available?

Reply:
ok - I see - this is just copying what was there previously